### PR TITLE
fix: create social signon user if doesn't already exist

### PR DESCRIPTION
**Pull request by danielgent**
Works for Google & Facebook!

Apple: will do on another PR :upside_down_face:

### CODE REVIEW

File: src/authentication-flows/social.ts [hunk 1]
```diff
@@ -117,9 +117,33 @@ export async function socialAuthCallback({
...
   const email = oauth2Profile.email.toLocaleLowerCase();
-  const user = await env.data.users.getByEmail(client.tenant_id, email);
+
+  // log out oauth2Profile so we know the structure of each...
+  // then we can pull out the id!
+
+  // this should actually be id! the social id pulled out before
+  let user = await env.data.users.getByEmail(client.tenant_id, email);
   if (!user) {
-    throw new HTTPException(403, { message: "User not found" });
```
Comment (markusahlstrand) [inline src/authentication-flows/social.ts:122]: Wow.. so now it's not possible to create a new user with social?

File: src/authentication-flows/social.ts [hunk 2]
```diff
@@ -117,9 +117,33 @@ export async function socialAuthCallback({
...
+    // throw new HTTPException(403, { message: "User not found" });
+    // we need to create the user!
+
+    user = await env.data.users.create(client.tenant_id, {
+      email,
+      tenant_id: client.tenant_id,
+      id: `email|${email}`,
+      name: email,
+      // ahhhh, this is the issue here. this is actually the bit we need to fix :-)
+      provider: "email",
```
Comment (markusahlstrand) [inline src/authentication-flows/social.ts:136]: You have this in `state.connection` above?

File: src/authentication-flows/social.ts [hunk 3]
```diff
@@ -116,9 +116,34 @@ export async function socialAuthCallback({
   const oauth2Profile = parseJwt(token.id_token!);
 
   const email = oauth2Profile.email.toLocaleLowerCase();
-  const user = await env.data.users.getByEmail(client.tenant_id, email);
+
+  // TODO - this should actually be id! the social id pulled out before
+  // we can fix once we've done account linking
```
Comment (danielgent) [inline src/authentication-flows/social.ts:121]: added comments but these are what we need to do imminently!

Now we've merged user linking I could give it a go

Here we would get by id and that would be the sub (so far! for Google & FB, let's see Apple :crossed_fingers: )

File: src/authentication-flows/social.ts [hunk 4]
```diff
@@ -116,9 +116,34 @@ export async function socialAuthCallback({
...
+  let user = await env.data.users.getByEmail(client.tenant_id, email);
```
Comment (markusahlstrand) [inline src/authentication-flows/social.ts:122]: Long term I guess we want to fetch the user by id here and handle linking later in the flow?

**Review by markusahlstrand — APPROVED** [review]
Looks good!

File: src/authentication-flows/social.ts (hunk 4 above)
Comment (danielgent) [reply src/authentication-flows/social.ts:122]: Exactly! (as per the comments)

Long-term meaning I could try this out now? :star_struck: 
Seems so close

**Pull request merged**